server/src/core/server/services/tenant/disposableEmails.ts

```diff
@@ -0,0 +1,53 @@
+import axios from "axios";
+import Logger from "bunyan";
+import { Parser } from "stream-json";
+import { streamArray } from "stream-json/streamers/StreamArray";
+
+import {
+  DISPOSABLE_EMAIL_DOMAIN_CACHE_DURATION,
+  DISPOSABLE_EMAIL_DOMAINS_LIST_URL,
+  DISPOSABLE_EMAIL_DOMAINS_REDIS_KEY,
+} from "coral-common/common/lib/constants";
+
+import { AugmentedRedis } from "../redis";
+
+export async function readDisposableEmailDomainsAndAddToRedis(
+  redis: AugmentedRedis,
+  logger: Logger
+) {
+  const now = new Date();
+
+  try {
+    const stream = await axios.get(`${DISPOSABLE_EMAIL_DOMAINS_LIST_URL}`, {
+      responseType: "stream",
+    });
+
+    const pipeline = stream.data.pipe(new Parser()).pipe(streamArray());
+
+    pipeline.on("data", async (data: { value: string }) => {
+      const domain = data.value;
+      await redis.set(
+        `${domain}${DISPOSABLE_EMAIL_DOMAINS_REDIS_KEY}`,
+        now.toISOString(),
+        "EX",
+        DISPOSABLE_EMAIL_DOMAIN_CACHE_DURATION
```

Comment:
if you set an expiry, there is a possibility of it being set, pods stay alive for a long while, then the keys disappear and we're hooped, there is no way to reload the domain lookup without a deployment 

I don't think this needs to be expired at all?

Reply:
All right, I updated to remove the expiry and use `hset` here.

Also fyi -- there is a button in the admin in the configuration section for this feature that can be clicked to update the disposable domains, so we shouldn't be in a situation where there's no way to reload without a deployment :)